### PR TITLE
Miscelaneous FHIRPersistenceJDBCImpl changes

### PR DESCRIPTION
**Pull request by lmsurpre**
1. refactored common logic from across create, update, and delete
methods into private helpers `copyAndSetResourceMetaFields` and
`createResourceDTO`

2. set the resourceBuilder to non-validating in
copyAndSetResourceMetaFields since the resource has already been
validated by this point

3. use try-with-catch for the GZIPOutputStream to ensure we never leak
resources

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Review by prb112 — APPROVED** [review]
LGTM - pretty straightforward

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java [hunk 1]
```diff
@@ -446,6 +429,42 @@ protected Action buildActionChain() {
...
+
+    @SuppressWarnings("unchecked")
+    private <T extends Resource> T copyAndSetResourceMetaFields(T resource, int newVersionNumber, Instant lastUpdated) {
+        Meta meta = resource.getMeta();
+        Meta.Builder metaBuilder = meta == null ? Meta.builder() : meta.toBuilder();
+        metaBuilder.versionId(Id.of(Integer.toString(newVersionNumber)));
+        metaBuilder.lastUpdated(lastUpdated);
+
+        Builder resourceBuilder = resource.toBuilder();
+        resourceBuilder.setValidating(false);
```
Comment (lmsurpre) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java:464]: setting this to false will prevent unnecessary parse validation for a resource that we already know was valid

**Review by JohnTimm — APPROVED** [review]
LGTM

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java [hunk 2]
```diff
@@ -446,6 +423,45 @@ protected Action buildActionChain() {
         }
     }
 
+    private <T extends Resource> com.ibm.fhir.persistence.jdbc.dto.Resource createResourceDTO(String logicalId, int newVersionNumber,
```
Comment (michaelwschroeder) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java:426]: javadoc for the new methods

**Review by michaelwschroeder — APPROVED** [review]
LGTM - one minor comment, and I don't think this should conflict with my changes for whole-system search

**Review by punktilious — APPROVED** [review]
LGTM

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java]: added

**Pull request merged**